src/llmq/quorums_signing_shares.cpp

```diff
@@ -1012,49 +1062,88 @@ bool CSigSharesManager::SendMessages()
 
         auto it1 = sigSessionAnnouncements.find(pnode->id);
         if (it1 != sigSessionAnnouncements.end()) {
+            std::vector<CSigSesAnn> msgs;
+            msgs.reserve(it1->second.size());
             for (auto& sigSesAnn : it1->second) {
                 LogPrint("llmq", "CSigSharesManager::SendMessages -- QSIGSESANN signHash=%s, sessionId=%d, node=%d\n",
                          CLLMQUtils::BuildSignHash(sigSesAnn).ToString(), sigSesAnn.sessionId, pnode->id);
-                g_connman->PushMessage(pnode, msgMaker.Make(NetMsgType::QSIGSESANN, sigSesAnn), false);
+                msgs.emplace_back(sigSesAnn);
+                if (msgs.size() >= MAX_MSGS_CNT_QSIGSESANN) {
+                    g_connman->PushMessage(pnode, msgMaker.Make(NetMsgType::QSIGSESANN, msgs), false);
+                    msgs.clear();
+                }
```

Comment:
```suggestion
                    didSend = true;
                }
```
and drop the one at line 1075. Same for all other msgs below.

Reply:
Not sure I understand...I moved the `didSend = true` lines out of the loops now.

Reply:
My idea was that `didSend` should be next to `PushMessage` where msg is actually sent.

Reply:
Ah ok, yeah would be semantically more correct, agree. I'll fix this.